src/hgraph/_operators/_operators.py

```diff
@@ -703,104 +692,69 @@ def my_len(ts: TS[MyType]) -> TS[int]:
 
 
 # SET Operators
-# TODO - these need to be merged into the binary ops we have in tss_operators
-# TODO - make this an operator and overload for #args
-
-def union(*args: TIME_SERIES_TYPE) -> TIME_SERIES_TYPE:
-    """
-    Performs a union of a collection of time-series values.
-    Each arg must be of the same type.
-    """
-    if len(args) == 1:
-        return args[0]
 
-    return union_tsl(TSL.from_ts(*args))
 
-
-# TODO - this can be an overload of union for unary TSL?
-# Also move to _tsl_operators
 @operator
-def union_tsl(tsl: TSL[TIME_SERIES_TYPE, SIZE]) -> TIME_SERIES_TYPE:
+def union(*tsl: TSL[TIME_SERIES_TYPE, SIZE], tp: Type[TIME_SERIES_TYPE] = AUTO_RESOLVE) -> TIME_SERIES_TYPE:
     """
     Performs a union of the provided time-series values.
 
-    By default, is ``reduce(union_op, tsl)``
+    By default, this is ``reduce(union_op, tsl)``
 
     Union is { p | p element of tsl[i] for i in range(len(tsl)) }
     """
-    # TODO - this doesn't seem to work without the zero()
-    return reduce(union_op, tsl)
-
-
-@operator
-def union_op(lhs: TIME_SERIES_TYPE, rhs: TIME_SERIES_TYPE) -> TIME_SERIES_TYPE:
-    """
-    Performs a union of the provided time-series values.
+    if len(tsl) == 1:
+        return tsl[0]
+    elif len(tsl) == 2:
+        return bit_or(tsl[0], tsl[1])
+    else:
+        return reduce(bit_or, tsl, zero(tp, bit_or))
 
-    Union is { p | p element of lhs and p element of rhs }
-    """
-    raise WiringError(
-        f"operator union_op is not implemented for {lhs.output_type} and {rhs.output_type}")
 
 
 @operator
-def str_(ts: TIME_SERIES_TYPE) -> TS[str]:
-    """
-    Implements python ```str()```
-    """
-    raise WiringError(f"operator str is not implemented for {ts.output_type}")
-
-
-def intersection(*args: TIME_SERIES_TYPE) -> TIME_SERIES_TYPE:
-    """
-    Performs a union of a collection of time-series values.
-    Each arg must be of the same type.
-    """
-    if len(args) == 1:
-        return args[0]
-
-    return intersection_tsl(TSL.from_ts(*args))
-
-
-@operator
-def intersection_tsl(tsl: TSL[TIME_SERIES_TYPE, SIZE]) -> TIME_SERIES_TYPE:
+def intersection(*tsl: TSL[TIME_SERIES_TYPE, SIZE], tp: Type[TIME_SERIES_TYPE] = AUTO_RESOLVE) -> TIME_SERIES_TYPE:
     """
```

Comment:
It seems we are combining definition of operator and implementation together, I though we were going to keep the operator as pure interface?